samples/azure/azure-service-fabric-routing/Core_6/CandidateVoteCount/CandidateVotes.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Threading.Tasks;
+using NServiceBus;
+
+public class CandidateVotes :
+    Saga<CandidateVotes.CandidateVoteData>,
+        IAmStartedByMessages<VotePlaced>,
+        IHandleMessages<CloseElection>,
+        IHandleMessages<TrackZipCodeReply>
+{
+
+    protected override void ConfigureHowToFindSaga(SagaPropertyMapper<CandidateVoteData> mapper)
+    {
+        mapper.ConfigureMapping<VotePlaced>(m => m.Candidate)
+            .ToSaga(s => s.Candidate);
+        mapper.ConfigureMapping<CloseElection>(m => m.Candidate)
+            .ToSaga(s => s.Candidate);
+    }
+
+    public Task Handle(VotePlaced message, IMessageHandlerContext context)
+    {
+        if (!Data.Started)
+        {
+            Data.Candidate = message.Candidate;
+            Data.Started = true;
+        }
+        Data.Count++;
+
+        var trackZipCode = new TrackZipCode
+        {
+            ZipCode = message.ZipCode
+        };
+        return context.Send(trackZipCode);
+    }
+
+    public async Task Handle(CloseElection message, IMessageHandlerContext context)
+    {
+        var reportVotes = new ReportVotes
+        {
+            Candidate = Data.Candidate,
+            NumberOfVotes = Data.Count
+        };
+        await context.SendLocal(reportVotes)
```

Comment:
I don't see ReportVotes being processed by the saga implying there is indirect coupling between the saga and the handler of ReportVotes as the saga is aware of the deployment of the handler.